lib/rules/strict-component-boundaries.js

```diff
@@ -0,0 +1,41 @@
+const pascalCase = require('pascal-case');
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Prevent module imports between components.',
+      category: 'Best Practises',
+      recommended: false,
+      uri:
+        'https://github.com/Shopify/eslint-plugin-shopify/blob/master/docs/rules/strict-component-boundaries.md',
+    },
+    fixable: null,
+  },
+  create(context) {
+    return {
+      ImportDeclaration(node) {
+        const pathParts = node.source.value
+          .split('/')
+          .filter((part) => part[0] !== '.');
+
+        if (
+          (hasAnotherComponentInPath(pathParts) && pathParts.length > 1) ||
+          (hasComponentDirectoryInPath(pathParts) && pathParts.length > 2)
```

Comment:
I think this might not catch ‘components/Foo’

Reply:
The first one should catch that https://github.com/Shopify/eslint-plugin-shopify/pull/98/files#diff-f91ed48b12fe78aaf97a3a26453bd862R31